### PR TITLE
Add orchagent heartbeat during warm-reboot UT

### PR DESCRIPTION
**Pull request by liuh-80**
Add orchagent heartbeat during warm-reboot UT

### Description of PR
Add orchagent heartbeat during warm-reboot UT

##### Work item tracking
- Microsoft ADO: 25295846

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Fix orchagent stuck error during warm-reboot:
https://github.com/sonic-net/sonic-buildimage/issues/16686

#### How did you do it?
Add new UT, freeze orchanget for warm-reboot then check the process listener not send alert.

#### How did you verify/test it?
Pass all UT

### CODE REVIEW

**Comment by liuh-80** [comment]
This PR depends on https://github.com/sonic-net/sonic-swss/pull/2923 merge and update submodule first

**Comment by liuh-80** [comment]
UT failed because t1 lag not have swss container, will improve UT:

{"changed": true, "cmd": "docker exec -i swss orchagent_restart_check", "delta": "0:00:00.022442", "end": "2023-11-14 03:54:23.198126", "failed": true, "msg": "non-zero return code", "rc": 1, "start": "2023-11-14 03:54:23.175684", "stderr": "Error response from daemon: No such container: swss", "stderr_lines": ["Error response from daemon: No such container: swss"], "stdout": "", "stdout_lines": []}

**Comment by ganglyu** [comment]
Can we us sonic mgmt test to verify this feature?

**Comment by liuh-80** [comment]
> Can we us sonic mgmt test to verify this feature?

warm-reboot will stuck orchagent so heartbeat message will not send, then proc-exit-listener will write stuck alert message to syslog.
This PR will do a e2e test to make sure the stuck alert does not trigger during warm-reboot.

This change difficult to add in sonic-swss repo UT. because the systemd will buffer heartbeat message, so test case in that repo very difficult to check the heartbeat message still sending during warm-reboot.

**Review by ganglyu — APPROVED** [review]

File: tests/process_monitoring/test_critical_process_monitoring.py [hunk 1]
```diff
@@ -609,3 +610,51 @@ def test_monitoring_critical_processes(duthosts, rand_one_dut_hostname, tbinfo,
     if not postcheck_critical_processes_status(duthost, up_bgp_neighbors):
         pytest.fail("Post-check failed after testing the process monitoring!")
     logger.info("Post-checking status of critical processes and BGP sessions was done!")
+
+
+def get_latest_stuck_message(duthost):
+    query_command = r"sudo sed -n 's/.*\(Process '\''orchagent'\'' is stuck in namespace\).*/\0/p'  /var/log/syslog"
```
Comment (ganglyu) [inline tests/process_monitoring/test_critical_process_monitoring.py:616]: We can use log analyzer

**Review by ganglyu — APPROVED** [review]

**Pull request merged**